### PR TITLE
[EAG2-130] Added PlayerState to program

### PR DESCRIPTION
**Pull request by RayWayy**

### CODE REVIEW

File: src/main/java/com/eleks/academy/whoami/core/SynchronousPlayer.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package com.eleks.academy.whoami.core;
 
 import com.eleks.academy.whoami.model.request.CharacterSuggestion;
+import com.eleks.academy.whoami.model.response.PlayerState;
```
Comment (VolodymyrKr-Eleks) [inline src/main/java/com/eleks/academy/whoami/core/SynchronousPlayer.java]: if we use PlayerState enum in core better to make it part of the core and move it here, right now as I see it is part of the models that we use in the core. It is ok when things are going out from core, but not good when core depends on things outside the core

File: src/main/java/com/eleks/academy/whoami/core/impl/PersistentPlayer.java [hunk 2]
```diff
@@ -2,16 +2,18 @@
 
 import com.eleks.academy.whoami.core.SynchronousPlayer;
 import com.eleks.academy.whoami.model.request.CharacterSuggestion;
+import com.eleks.academy.whoami.model.response.PlayerState;
 import lombok.EqualsAndHashCode;
 
 import java.util.Objects;
 
-@EqualsAndHashCode
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
```
Comment (VolodymyrKr-Eleks) [inline src/main/java/com/eleks/academy/whoami/core/impl/PersistentPlayer.java]: why do we need `onlyExplicitlyIncluded = true` if no one class field is included, that means that in hashCode and equals there are no fields to check, for our purposes it's enough just to have `@EqualsAndHashCode` as it was before

File: src/main/java/com/eleks/academy/whoami/core/impl/PersistentPlayer.java [hunk 3]
```diff
@@ -32,10 +34,21 @@ public String getCharacter() {
...
+	@Override
+	public PlayerState getPlayerState() {
+		return this.playerState;
+	}
+
 	@Override
 	public void suggestCharacter(CharacterSuggestion suggestion) {
 		this.nickName = suggestion.getNickName();
 		this.character = suggestion.getCharacter();
+		this.setPlayerState(PlayerState.READY);
```
Comment (VolodymyrKr-Eleks) [inline src/main/java/com/eleks/academy/whoami/core/impl/PersistentPlayer.java]: you have direct access to the field inside the class why do we need to use setter?

File: src/main/java/com/eleks/academy/whoami/core/state/ProcessingQuestion.java [hunk 4]
```diff
@@ -36,4 +38,14 @@ public String getCurrentTurn() {
 		return this.currentPlayer;
 	}
 
+	private void updatePlayerStates() {
```
Comment (VolodymyrKr-Eleks) [inline src/main/java/com/eleks/academy/whoami/core/state/ProcessingQuestion.java]: for now, as we see that this action about setting the state to players happens only in the constructor and nowhere else, let's keep it there.
Private methods mostly are needed when there is some common code within the class that repeats or an action that provides a single result takes many lines of code to be reached. In such cases, private methods are needed, as you may see here we do not have such cases - so no need of private method

**Review by VolodymyrKr-Eleks — CHANGES_REQUESTED** [review]

File: src/main/java/com/eleks/academy/whoami/core/state/ProcessingQuestion.java (hunk 4 above)
Comment (RayWayy) [reply src/main/java/com/eleks/academy/whoami/core/state/ProcessingQuestion.java]: Resolved
Understand

File: src/main/java/com/eleks/academy/whoami/core/impl/PersistentPlayer.java (hunk 3 above)
Comment (RayWayy) [reply src/main/java/com/eleks/academy/whoami/core/impl/PersistentPlayer.java]: Resolved

File: src/main/java/com/eleks/academy/whoami/core/SynchronousPlayer.java (hunk 1 above)
Comment (RayWayy) [reply src/main/java/com/eleks/academy/whoami/core/SynchronousPlayer.java]: Resolved

File: src/main/java/com/eleks/academy/whoami/core/impl/PersistentPlayer.java (hunk 2 above)
Comment (RayWayy) [reply src/main/java/com/eleks/academy/whoami/core/impl/PersistentPlayer.java]: Resolved
In this line after adding `onlyExplicitlyIncluded = true` test are passed. So I add and tests are worked. I remove this line and find in tests where was mistake. So it was fixed

**Pull request merged**